{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Merging the pairs\n",
    "\n",
    "`bbmerge` package was used in this step"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LM1 LM4 LM5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "java -ea -Xmx24675m -Xms24675m -cp /usr/local/bin/bbmap/current/ jgi.BBMerge in=LM1_R1.fastq in2=LM1_R2.fastq out=corrected_LM1_R1.fastq out2=corrected_LM1_R2.fastq ecco=t mix=t\n",
      "Executing jgi.BBMerge [in=LM1_R1.fastq, in2=LM1_R2.fastq, out=corrected_LM1_R1.fastq, out2=corrected_LM1_R2.fastq, ecco=t, mix=t]\n",
      "Version 38.86\n",
      "\n",
      "Writing mergable reads unmerged in two files.\n",
      "Started output threads.\n",
      "Total time: 93.728 seconds.\n",
      "\n",
      "Pairs:               \t34375177\n",
      "Joined:              \t16325022   \t47.491%\n",
      "Ambiguous:           \t18050147   \t52.509%\n",
      "No Solution:         \t8       \t0.000%\n",
      "Too Short:           \t0       \t0.000%\n",
      "Errors Corrected:    \t552636\n",
      "\n",
      "Avg Insert:          \t59.3\n",
      "Standard Deviation:  \t9.5\n",
      "Mode:                \t70\n",
      "\n",
      "Insert range:        \t35 - 74\n",
      "90th percentile:     \t71\n",
      "75th percentile:     \t67\n",
      "50th percentile:     \t61\n",
      "25th percentile:     \t53\n",
      "10th percentile:     \t45\n",
      "java -ea -Xmx24520m -Xms24520m -cp /usr/local/bin/bbmap/current/ jgi.BBMerge minoverlap=6 mininsert=20 mininsert0=20 in=corrected_LM1_R1.fastq in2=corrected_LM1_R2.fastq out=merged_LM1.fastq outu=unmerged_LM1_R1.fastq outu2=unmerged_LM1_R2.fastq\n",
      "Executing jgi.BBMerge [minoverlap=6, mininsert=20, mininsert0=20, in=corrected_LM1_R1.fastq, in2=corrected_LM1_R2.fastq, out=merged_LM1.fastq, outu=unmerged_LM1_R1.fastq, outu2=unmerged_LM1_R2.fastq]\n",
      "Version 38.86\n",
      "\n",
      "Writing mergable reads merged.\n",
      "Started output threads.\n",
      "Total time: 71.500 seconds.\n",
      "\n",
      "Pairs:               \t34375177\n",
      "Joined:              \t17284830   \t50.283%\n",
      "Ambiguous:           \t17090339   \t49.717%\n",
      "No Solution:         \t8       \t0.000%\n",
      "Too Short:           \t0       \t0.000%\n",
      "\n",
      "Avg Insert:          \t57.7\n",
      "Standard Deviation:  \t11.4\n",
      "Mode:                \t70\n",
      "\n",
      "Insert range:        \t20 - 74\n",
      "90th percentile:     \t71\n",
      "75th percentile:     \t67\n",
      "50th percentile:     \t60\n",
      "25th percentile:     \t51\n",
      "10th percentile:     \t41\n",
      "java -ea -Xmx24577m -Xms24577m -cp /usr/local/bin/bbmap/current/ jgi.BBMerge in=LM4_R1.fastq in2=LM4_R2.fastq out=corrected_LM4_R1.fastq out2=corrected_LM4_R2.fastq ecco=t mix=t\n",
      "Executing jgi.BBMerge [in=LM4_R1.fastq, in2=LM4_R2.fastq, out=corrected_LM4_R1.fastq, out2=corrected_LM4_R2.fastq, ecco=t, mix=t]\n",
      "Version 38.86\n",
      "\n",
      "Writing mergable reads unmerged in two files.\n",
      "Started output threads.\n",
      "Total time: 107.425 seconds.\n",
      "\n",
      "Pairs:               \t35117715\n",
      "Joined:              \t17963712   \t51.153%\n",
      "Ambiguous:           \t17153997   \t48.847%\n",
      "No Solution:         \t6       \t0.000%\n",
      "Too Short:           \t0       \t0.000%\n",
      "Errors Corrected:    \t608187\n",
      "\n",
      "Avg Insert:          \t58.9\n",
      "Standard Deviation:  \t9.8\n",
      "Mode:                \t70\n",
      "\n",
      "Insert range:        \t35 - 74\n",
      "90th percentile:     \t71\n",
      "75th percentile:     \t67\n",
      "50th percentile:     \t60\n",
      "25th percentile:     \t52\n",
      "10th percentile:     \t44\n",
      "java -ea -Xmx24569m -Xms24569m -cp /usr/local/bin/bbmap/current/ jgi.BBMerge minoverlap=6 mininsert=20 mininsert0=20 in=corrected_LM4_R1.fastq in2=corrected_LM4_R2.fastq out=merged_LM4.fastq outu=unmerged_LM4_R1.fastq outu2=unmerged_LM4_R2.fastq\n",
      "Executing jgi.BBMerge [minoverlap=6, mininsert=20, mininsert0=20, in=corrected_LM4_R1.fastq, in2=corrected_LM4_R2.fastq, out=merged_LM4.fastq, outu=unmerged_LM4_R1.fastq, outu2=unmerged_LM4_R2.fastq]\n",
      "Version 38.86\n",
      "\n",
      "Writing mergable reads merged.\n",
      "Started output threads.\n",
      "Total time: 57.610 seconds.\n",
      "\n",
      "Pairs:               \t35117715\n",
      "Joined:              \t19397028   \t55.234%\n",
      "Ambiguous:           \t15720681   \t44.766%\n",
      "No Solution:         \t6       \t0.000%\n",
      "Too Short:           \t0       \t0.000%\n",
      "\n",
      "Avg Insert:          \t56.8\n",
      "Standard Deviation:  \t12.1\n",
      "Mode:                \t30\n",
      "\n",
      "Insert range:        \t20 - 74\n",
      "90th percentile:     \t70\n",
      "75th percentile:     \t67\n",
      "50th percentile:     \t59\n",
      "25th percentile:     \t50\n",
      "10th percentile:     \t38\n",
      "java -ea -Xmx24598m -Xms24598m -cp /usr/local/bin/bbmap/current/ jgi.BBMerge in=LM5_R1.fastq in2=LM5_R2.fastq out=corrected_LM5_R1.fastq out2=corrected_LM5_R2.fastq ecco=t mix=t\n",
      "Executing jgi.BBMerge [in=LM5_R1.fastq, in2=LM5_R2.fastq, out=corrected_LM5_R1.fastq, out2=corrected_LM5_R2.fastq, ecco=t, mix=t]\n",
      "Version 38.86\n",
      "\n",
      "Writing mergable reads unmerged in two files.\n",
      "Started output threads.\n",
      "Total time: 90.264 seconds.\n",
      "\n",
      "Pairs:               \t29123746\n",
      "Joined:              \t14443168   \t49.592%\n",
      "Ambiguous:           \t14680570   \t50.408%\n",
      "No Solution:         \t8       \t0.000%\n",
      "Too Short:           \t0       \t0.000%\n",
      "Errors Corrected:    \t526069\n",
      "\n",
      "Avg Insert:          \t59.2\n",
      "Standard Deviation:  \t9.5\n",
      "Mode:                \t70\n",
      "\n",
      "Insert range:        \t35 - 74\n",
      "90th percentile:     \t71\n",
      "75th percentile:     \t67\n",
      "50th percentile:     \t60\n",
      "25th percentile:     \t52\n",
      "10th percentile:     \t46\n",
      "java -ea -Xmx24567m -Xms24567m -cp /usr/local/bin/bbmap/current/ jgi.BBMerge minoverlap=6 mininsert=20 mininsert0=20 in=corrected_LM5_R1.fastq in2=corrected_LM5_R2.fastq out=merged_LM5.fastq outu=unmerged_LM5_R1.fastq outu2=unmerged_LM5_R2.fastq\n",
      "Executing jgi.BBMerge [minoverlap=6, mininsert=20, mininsert0=20, in=corrected_LM5_R1.fastq, in2=corrected_LM5_R2.fastq, out=merged_LM5.fastq, outu=unmerged_LM5_R1.fastq, outu2=unmerged_LM5_R2.fastq]\n",
      "Version 38.86\n",
      "\n",
      "Writing mergable reads merged.\n",
      "Started output threads.\n",
      "Total time: 53.480 seconds.\n",
      "\n",
      "Pairs:               \t29123746\n",
      "Joined:              \t15527011   \t53.314%\n",
      "Ambiguous:           \t13596727   \t46.686%\n",
      "No Solution:         \t8       \t0.000%\n",
      "Too Short:           \t0       \t0.000%\n",
      "\n",
      "Avg Insert:          \t57.2\n",
      "Standard Deviation:  \t11.8\n",
      "Mode:                \t30\n",
      "\n",
      "Insert range:        \t20 - 74\n",
      "90th percentile:     \t71\n",
      "75th percentile:     \t67\n",
      "50th percentile:     \t59\n",
      "25th percentile:     \t51\n",
      "10th percentile:     \t39\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "input_list_file=\"sample_list.txt\"\n",
    "while IFS='' read -r line || [[ -n \"$line\" ]]; do\n",
    "  sample_list=\"$sample_list $line\"\n",
    "done <\"$input_list_file\"\n",
    "echo $sample_list\n",
    "\n",
    "for read in $sample_list; do\n",
    "    bbmerge-auto.sh in=$read'_R1.fastq' \\\n",
    "               in2=$read'_R2.fastq' \\\n",
    "               out='corrected_'$read'_R1.fastq' \\\n",
    "               out2='corrected_'$read'_R2.fastq' \\\n",
    "               ecco=t \\\n",
    "               mix=t\n",
    "\n",
    "    bbmerge-auto.sh minoverlap=6 \\\n",
    "                    mininsert=20 \\\n",
    "                    mininsert0=20 \\\n",
    "                    in='corrected_'$read'_R1.fastq' \\\n",
    "                    in2='corrected_'$read'_R2.fastq' \\\n",
    "                    out=\"merged_\"$read\".fastq\" \\\n",
    "                    outu=\"unmerged_\"$read\"_R1.fastq\" \\\n",
    "                    outu2=\"unmerged_\"$read\"_R2.fastq\" \n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Move UMI to read name and Process with merged reads only\n",
    "\n",
    "`cutadapt`, as well as `move_umi.py` from https://github.com/yz201906/RibOxi-seq was used in this step"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LM1 LM4 LM5\n",
      "This is cutadapt 2.10 with Python 3.8.5\n",
      "Command line parameters: -j 12 --discard-untrimmed --minimum-length 20 -a CTGTAGGCACCATCAATGAC -e 0.1 -o dt_LM1.fastq LM1.fastq\n",
      "Processing reads on 12 cores in single-end mode ...\n",
      "Finished in 31.64 s (1 us/read; 65.19 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              34,375,177\n",
      "Reads with adapters:                17,854,174 (51.9%)\n",
      "Reads written (passing filters):    16,214,824 (47.2%)\n",
      "\n",
      "Total basepairs processed: 1,714,674,384 bp\n",
      "Total written (filtered):    648,257,920 bp (37.8%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTGTAGGCACCATCAATGAC; Type: regular 3'; Length: 20; Trimmed: 17854174 times\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 2.3%\n",
      "  C: 0.5%\n",
      "  G: 96.2%\n",
      "  T: 1.0%\n",
      "  none/other: 0.0%\n",
      "WARNING:\n",
      "    The adapter is preceded by \"G\" extremely often.\n",
      "    The provided adapter sequence could be incomplete at its 3' end.\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t280708\t537112.1\t0\t280708\n",
      "4\t68914\t134278.0\t0\t68914\n",
      "5\t25404\t33569.5\t0\t25404\n",
      "6\t21879\t8392.4\t0\t21879\n",
      "7\t14225\t2098.1\t0\t14225\n",
      "8\t12698\t524.5\t0\t12698\n",
      "9\t12453\t131.1\t0\t11799 654\n",
      "10\t13134\t32.8\t1\t10138 2996\n",
      "11\t18624\t8.2\t1\t14800 3824\n",
      "12\t12937\t2.0\t1\t10183 2754\n",
      "13\t9898\t0.5\t1\t7494 2404\n",
      "14\t8736\t0.1\t1\t6460 2276\n",
      "15\t8505\t0.0\t1\t6787 1718\n",
      "16\t8517\t0.0\t1\t6595 1922\n",
      "17\t7031\t0.0\t1\t5477 1554\n",
      "18\t9236\t0.0\t1\t5839 1699 1698\n",
      "19\t35814\t0.0\t1\t6098 28153 1563\n",
      "20\t17194903\t0.0\t2\t16522946 581570 90387\n",
      "21\t12262\t0.0\t2\t4803 6665 794\n",
      "22\t4779\t0.0\t2\t3367 866 546\n",
      "23\t4641\t0.0\t2\t3282 893 466\n",
      "24\t4523\t0.0\t2\t3248 803 472\n",
      "25\t5388\t0.0\t2\t3890 1017 481\n",
      "26\t7207\t0.0\t2\t5304 1357 546\n",
      "27\t5094\t0.0\t2\t3696 965 433\n",
      "28\t4795\t0.0\t2\t3600 847 348\n",
      "29\t1949\t0.0\t2\t1391 390 168\n",
      "30\t308\t0.0\t2\t213 69 26\n",
      "31\t331\t0.0\t2\t236 65 30\n",
      "32\t31457\t0.0\t2\t23512 5474 2471\n",
      "33\t1867\t0.0\t2\t1325 371 171\n",
      "34\t312\t0.0\t2\t214 66 32\n",
      "35\t179\t0.0\t2\t124 39 16\n",
      "36\t111\t0.0\t2\t71 29 11\n",
      "37\t55\t0.0\t2\t31 16 8\n",
      "38\t62\t0.0\t2\t37 21 4\n",
      "39\t44\t0.0\t2\t24 13 7\n",
      "40\t68\t0.0\t2\t38 23 7\n",
      "41\t27\t0.0\t2\t9 6 12\n",
      "42\t5009\t0.0\t2\t22 64 4923\n",
      "43\t2\t0.0\t2\t2\n",
      "44\t3\t0.0\t2\t3\n",
      "45\t1\t0.0\t2\t1\n",
      "46\t2\t0.0\t2\t2\n",
      "47\t1\t0.0\t2\t0 1\n",
      "48\t51\t0.0\t2\t51\n",
      "49\t15\t0.0\t2\t15\n",
      "50\t13\t0.0\t2\t13\n",
      "51\t1\t0.0\t2\t1\n",
      "56\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "WARNING:\n",
      "    One or more of your adapter sequences may be incomplete.\n",
      "    Please see the detailed output above.\n",
      "Number of merged reads as input:16214824\n",
      "Number of non-mispriming reads:16214824\n",
      "% passed reads:100.0%\n",
      "This is cutadapt 2.10 with Python 3.8.5\n",
      "Command line parameters: -j 12 --discard-untrimmed --minimum-length 20 -a ATCACG -e 0.01 -o trimmed_LM1.fastq umiRemoved_LM1_R1.fastq\n",
      "Processing reads on 12 cores in single-end mode ...\n",
      "Finished in 15.80 s (1 us/read; 61.58 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              16,214,824\n",
      "Reads with adapters:                15,154,274 (93.5%)\n",
      "Reads written (passing filters):    13,155,650 (81.1%)\n",
      "\n",
      "Total basepairs processed:   583,398,624 bp\n",
      "Total written (filtered):    425,369,511 bp (72.9%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: ATCACG; Type: regular 3'; Length: 6; Trimmed: 15154274 times\n",
      "\n",
      "No. of allowed errors:\n",
      "0-6 bp: 0\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 36.0%\n",
      "  C: 18.0%\n",
      "  G: 21.0%\n",
      "  T: 25.0%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t2195\t253356.6\t0\t2195\n",
      "4\t1761\t63339.2\t0\t1761\n",
      "5\t3370\t15834.8\t0\t3370\n",
      "6\t15140251\t3958.7\t0\t15140251\n",
      "7\t3576\t3958.7\t0\t3576\n",
      "8\t7\t3958.7\t0\t7\n",
      "10\t1\t3958.7\t0\t1\n",
      "11\t5\t3958.7\t0\t5\n",
      "12\t78\t3958.7\t0\t78\n",
      "13\t50\t3958.7\t0\t50\n",
      "14\t80\t3958.7\t0\t80\n",
      "15\t68\t3958.7\t0\t68\n",
      "16\t98\t3958.7\t0\t98\n",
      "17\t119\t3958.7\t0\t119\n",
      "18\t109\t3958.7\t0\t109\n",
      "19\t127\t3958.7\t0\t127\n",
      "20\t89\t3958.7\t0\t89\n",
      "21\t305\t3958.7\t0\t305\n",
      "22\t88\t3958.7\t0\t88\n",
      "23\t78\t3958.7\t0\t78\n",
      "24\t118\t3958.7\t0\t118\n",
      "25\t93\t3958.7\t0\t93\n",
      "26\t112\t3958.7\t0\t112\n",
      "27\t90\t3958.7\t0\t90\n",
      "28\t110\t3958.7\t0\t110\n",
      "29\t79\t3958.7\t0\t79\n",
      "30\t132\t3958.7\t0\t132\n",
      "31\t68\t3958.7\t0\t68\n",
      "32\t148\t3958.7\t0\t148\n",
      "33\t150\t3958.7\t0\t150\n",
      "34\t120\t3958.7\t0\t120\n",
      "35\t85\t3958.7\t0\t85\n",
      "36\t46\t3958.7\t0\t46\n",
      "37\t43\t3958.7\t0\t43\n",
      "38\t40\t3958.7\t0\t40\n",
      "39\t41\t3958.7\t0\t41\n",
      "40\t25\t3958.7\t0\t25\n",
      "41\t105\t3958.7\t0\t105\n",
      "42\t49\t3958.7\t0\t49\n",
      "43\t33\t3958.7\t0\t33\n",
      "44\t45\t3958.7\t0\t45\n",
      "45\t28\t3958.7\t0\t28\n",
      "46\t28\t3958.7\t0\t28\n",
      "47\t14\t3958.7\t0\t14\n",
      "48\t12\t3958.7\t0\t12\n",
      "49\t5\t3958.7\t0\t5\n",
      "This is cutadapt 2.10 with Python 3.8.5\n",
      "Command line parameters: -j 12 --discard-untrimmed --minimum-length 20 -a CTGTAGGCACCATCAATGAC -e 0.1 -o dt_LM4.fastq LM4.fastq\n",
      "Processing reads on 12 cores in single-end mode ...\n",
      "Finished in 33.20 s (1 us/read; 63.46 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              35,117,715\n",
      "Reads with adapters:                19,973,186 (56.9%)\n",
      "Reads written (passing filters):    17,644,632 (50.2%)\n",
      "\n",
      "Total basepairs processed: 1,762,109,735 bp\n",
      "Total written (filtered):    702,234,118 bp (39.9%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTGTAGGCACCATCAATGAC; Type: regular 3'; Length: 20; Trimmed: 19973186 times\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 2.6%\n",
      "  C: 0.5%\n",
      "  G: 95.7%\n",
      "  T: 1.1%\n",
      "  none/other: 0.0%\n",
      "WARNING:\n",
      "    The adapter is preceded by \"G\" extremely often.\n",
      "    The provided adapter sequence could be incomplete at its 3' end.\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t266370\t548714.3\t0\t266370\n",
      "4\t59289\t137178.6\t0\t59289\n",
      "5\t23129\t34294.6\t0\t23129\n",
      "6\t21649\t8573.7\t0\t21649\n",
      "7\t14141\t2143.4\t0\t14141\n",
      "8\t12222\t535.9\t0\t12222\n",
      "9\t12694\t134.0\t0\t12039 655\n",
      "10\t13166\t33.5\t1\t10293 2873\n",
      "11\t20939\t8.4\t1\t16664 4275\n",
      "12\t13209\t2.1\t1\t10336 2873\n",
      "13\t10416\t0.5\t1\t7874 2542\n",
      "14\t9070\t0.1\t1\t6751 2319\n",
      "15\t9037\t0.0\t1\t7119 1918\n",
      "16\t9077\t0.0\t1\t7088 1989\n",
      "17\t7751\t0.0\t1\t6045 1706\n",
      "18\t10637\t0.0\t1\t6531 1907 2199\n",
      "19\t43045\t0.0\t1\t7159 34039 1847\n",
      "20\t19298514\t0.0\t2\t18578400 627552 92562\n",
      "21\t15514\t0.0\t2\t6026 8508 980\n",
      "22\t5597\t0.0\t2\t3889 1103 605\n",
      "23\t5849\t0.0\t2\t4171 1092 586\n",
      "24\t5780\t0.0\t2\t4163 1064 553\n",
      "25\t6764\t0.0\t2\t4943 1241 580\n",
      "26\t9130\t0.0\t2\t6748 1667 715\n",
      "27\t5758\t0.0\t2\t4183 1066 509\n",
      "28\t4181\t0.0\t2\t3140 730 311\n",
      "29\t1577\t0.0\t2\t1138 319 120\n",
      "30\t262\t0.0\t2\t157 63 42\n",
      "31\t425\t0.0\t2\t311 85 29\n",
      "32\t48081\t0.0\t2\t35945 8587 3549\n",
      "33\t2903\t0.0\t2\t2092 559 252\n",
      "34\t413\t0.0\t2\t282 100 31\n",
      "35\t262\t0.0\t2\t186 51 25\n",
      "36\t151\t0.0\t2\t101 29 21\n",
      "37\t45\t0.0\t2\t32 11 2\n",
      "38\t117\t0.0\t2\t66 40 11\n",
      "39\t52\t0.0\t2\t32 10 10\n",
      "40\t88\t0.0\t2\t50 31 7\n",
      "41\t53\t0.0\t2\t26 15 12\n",
      "42\t5710\t0.0\t2\t38 81 5591\n",
      "43\t2\t0.0\t2\t1 1\n",
      "44\t1\t0.0\t2\t1\n",
      "45\t6\t0.0\t2\t5 1\n",
      "46\t2\t0.0\t2\t2\n",
      "47\t3\t0.0\t2\t3\n",
      "48\t72\t0.0\t2\t72\n",
      "49\t16\t0.0\t2\t16\n",
      "50\t14\t0.0\t2\t14\n",
      "55\t2\t0.0\t2\t2\n",
      "60\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "WARNING:\n",
      "    One or more of your adapter sequences may be incomplete.\n",
      "    Please see the detailed output above.\n",
      "Number of merged reads as input:17644632\n",
      "Number of non-mispriming reads:17644632\n",
      "% passed reads:100.0%\n",
      "This is cutadapt 2.10 with Python 3.8.5\n",
      "Command line parameters: -j 12 --discard-untrimmed --minimum-length 20 -a ATCACG -e 0.01 -o trimmed_LM4.fastq umiRemoved_LM4_R1.fastq\n",
      "Processing reads on 12 cores in single-end mode ...\n",
      "Finished in 17.14 s (1 us/read; 61.77 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              17,644,632\n",
      "Reads with adapters:                16,394,045 (92.9%)\n",
      "Reads written (passing filters):    14,094,368 (79.9%)\n",
      "\n",
      "Total basepairs processed:   631,655,590 bp\n",
      "Total written (filtered):    455,727,867 bp (72.1%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: ATCACG; Type: regular 3'; Length: 6; Trimmed: 16394045 times\n",
      "\n",
      "No. of allowed errors:\n",
      "0-6 bp: 0\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 37.5%\n",
      "  C: 17.8%\n",
      "  G: 20.8%\n",
      "  T: 23.9%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t1674\t275697.4\t0\t1674\n",
      "4\t1698\t68924.3\t0\t1698\n",
      "5\t3776\t17231.1\t0\t3776\n",
      "6\t16379686\t4307.8\t0\t16379686\n",
      "7\t4232\t4307.8\t0\t4232\n",
      "8\t9\t4307.8\t0\t9\n",
      "9\t3\t4307.8\t0\t3\n",
      "10\t3\t4307.8\t0\t3\n",
      "11\t5\t4307.8\t0\t5\n",
      "12\t73\t4307.8\t0\t73\n",
      "13\t60\t4307.8\t0\t60\n",
      "14\t91\t4307.8\t0\t91\n",
      "15\t73\t4307.8\t0\t73\n",
      "16\t80\t4307.8\t0\t80\n",
      "17\t86\t4307.8\t0\t86\n",
      "18\t125\t4307.8\t0\t125\n",
      "19\t136\t4307.8\t0\t136\n",
      "20\t121\t4307.8\t0\t121\n",
      "21\t322\t4307.8\t0\t322\n",
      "22\t72\t4307.8\t0\t72\n",
      "23\t84\t4307.8\t0\t84\n",
      "24\t118\t4307.8\t0\t118\n",
      "25\t103\t4307.8\t0\t103\n",
      "26\t72\t4307.8\t0\t72\n",
      "27\t89\t4307.8\t0\t89\n",
      "28\t83\t4307.8\t0\t83\n",
      "29\t74\t4307.8\t0\t74\n",
      "30\t113\t4307.8\t0\t113\n",
      "31\t70\t4307.8\t0\t70\n",
      "32\t133\t4307.8\t0\t133\n",
      "33\t124\t4307.8\t0\t124\n",
      "34\t111\t4307.8\t0\t111\n",
      "35\t90\t4307.8\t0\t90\n",
      "36\t40\t4307.8\t0\t40\n",
      "37\t32\t4307.8\t0\t32\n",
      "38\t30\t4307.8\t0\t30\n",
      "39\t39\t4307.8\t0\t39\n",
      "40\t19\t4307.8\t0\t19\n",
      "41\t100\t4307.8\t0\t100\n",
      "42\t34\t4307.8\t0\t34\n",
      "43\t26\t4307.8\t0\t26\n",
      "44\t42\t4307.8\t0\t42\n",
      "45\t28\t4307.8\t0\t28\n",
      "46\t33\t4307.8\t0\t33\n",
      "47\t10\t4307.8\t0\t10\n",
      "48\t22\t4307.8\t0\t22\n",
      "49\t1\t4307.8\t0\t1\n",
      "This is cutadapt 2.10 with Python 3.8.5\n",
      "Command line parameters: -j 12 --discard-untrimmed --minimum-length 20 -a CTGTAGGCACCATCAATGAC -e 0.1 -o dt_LM5.fastq LM5.fastq\n",
      "Processing reads on 12 cores in single-end mode ...\n",
      "Finished in 26.92 s (1 us/read; 64.92 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              29,123,746\n",
      "Reads with adapters:                16,102,950 (55.3%)\n",
      "Reads written (passing filters):    14,444,394 (49.6%)\n",
      "\n",
      "Total basepairs processed: 1,459,191,216 bp\n",
      "Total written (filtered):    575,647,485 bp (39.4%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTGTAGGCACCATCAATGAC; Type: regular 3'; Length: 20; Trimmed: 16102950 times\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-19 bp: 1; 20 bp: 2\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 2.1%\n",
      "  C: 0.5%\n",
      "  G: 96.3%\n",
      "  T: 1.0%\n",
      "  none/other: 0.0%\n",
      "WARNING:\n",
      "    The adapter is preceded by \"G\" extremely often.\n",
      "    The provided adapter sequence could be incomplete at its 3' end.\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t306186\t455058.5\t0\t306186\n",
      "4\t52429\t113764.6\t0\t52429\n",
      "5\t21062\t28441.2\t0\t21062\n",
      "6\t17318\t7110.3\t0\t17318\n",
      "7\t12719\t1777.6\t0\t12719\n",
      "8\t12968\t444.4\t0\t12968\n",
      "9\t12200\t111.1\t0\t11587 613\n",
      "10\t13537\t27.8\t1\t10536 3001\n",
      "11\t21647\t6.9\t1\t17273 4374\n",
      "12\t12734\t1.7\t1\t10038 2696\n",
      "13\t9708\t0.4\t1\t7341 2367\n",
      "14\t8509\t0.1\t1\t6325 2184\n",
      "15\t8330\t0.0\t1\t6593 1737\n",
      "16\t7757\t0.0\t1\t5945 1812\n",
      "17\t6241\t0.0\t1\t4804 1437\n",
      "18\t8429\t0.0\t1\t5522 1722 1185\n",
      "19\t30384\t0.0\t1\t5931 23129 1324\n",
      "20\t15445708\t0.0\t2\t14854075 506967 84666\n",
      "21\t12164\t0.0\t2\t4681 6687 796\n",
      "22\t4342\t0.0\t2\t2999 855 488\n",
      "23\t4497\t0.0\t2\t3228 830 439\n",
      "24\t4377\t0.0\t2\t3126 822 429\n",
      "25\t4975\t0.0\t2\t3615 914 446\n",
      "26\t6174\t0.0\t2\t4561 1127 486\n",
      "27\t3578\t0.0\t2\t2570 696 312\n",
      "28\t2116\t0.0\t2\t1578 374 164\n",
      "29\t908\t0.0\t2\t668 181 59\n",
      "30\t190\t0.0\t2\t128 45 17\n",
      "31\t392\t0.0\t2\t284 83 25\n",
      "32\t43821\t0.0\t2\t32763 7557 3501\n",
      "33\t2657\t0.0\t2\t1904 527 226\n",
      "34\t367\t0.0\t2\t254 81 32\n",
      "35\t207\t0.0\t2\t148 44 15\n",
      "36\t141\t0.0\t2\t92 27 22\n",
      "37\t36\t0.0\t2\t26 8 2\n",
      "38\t89\t0.0\t2\t54 16 19\n",
      "39\t33\t0.0\t2\t20 8 5\n",
      "40\t60\t0.0\t2\t38 14 8\n",
      "41\t33\t0.0\t2\t11 4 18\n",
      "42\t3895\t0.0\t2\t28 40 3827\n",
      "43\t1\t0.0\t2\t1\n",
      "45\t1\t0.0\t2\t1\n",
      "46\t1\t0.0\t2\t0 1\n",
      "48\t16\t0.0\t2\t15 1\n",
      "49\t7\t0.0\t2\t7\n",
      "50\t4\t0.0\t2\t4\n",
      "52\t1\t0.0\t2\t1\n",
      "53\t1\t0.0\t2\t1\n",
      "\n",
      "\n",
      "WARNING:\n",
      "    One or more of your adapter sequences may be incomplete.\n",
      "    Please see the detailed output above.\n",
      "Number of merged reads as input:14444394\n",
      "Number of non-mispriming reads:14444394\n",
      "% passed reads:100.0%\n",
      "This is cutadapt 2.10 with Python 3.8.5\n",
      "Command line parameters: -j 12 --discard-untrimmed --minimum-length 20 -a ATCACG -e 0.01 -o trimmed_LM5.fastq umiRemoved_LM5_R1.fastq\n",
      "Processing reads on 12 cores in single-end mode ...\n",
      "Finished in 14.49 s (1 us/read; 59.81 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              14,444,394\n",
      "Reads with adapters:                13,474,321 (93.3%)\n",
      "Reads written (passing filters):    11,709,937 (81.1%)\n",
      "\n",
      "Total basepairs processed:   517,869,909 bp\n",
      "Total written (filtered):    376,772,202 bp (72.8%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: ATCACG; Type: regular 3'; Length: 6; Trimmed: 13474321 times\n",
      "\n",
      "No. of allowed errors:\n",
      "0-6 bp: 0\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 36.6%\n",
      "  C: 20.1%\n",
      "  G: 20.5%\n",
      "  T: 22.8%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t1099\t225693.7\t0\t1099\n",
      "4\t1234\t56423.4\t0\t1234\n",
      "5\t3057\t14105.9\t0\t3057\n",
      "6\t13462089\t3526.5\t0\t13462089\n",
      "7\t3415\t3526.5\t0\t3415\n",
      "8\t2\t3526.5\t0\t2\n",
      "9\t2\t3526.5\t0\t2\n",
      "10\t3\t3526.5\t0\t3\n",
      "11\t4\t3526.5\t0\t4\n",
      "12\t65\t3526.5\t0\t65\n",
      "13\t57\t3526.5\t0\t57\n",
      "14\t78\t3526.5\t0\t78\n",
      "15\t51\t3526.5\t0\t51\n",
      "16\t84\t3526.5\t0\t84\n",
      "17\t108\t3526.5\t0\t108\n",
      "18\t144\t3526.5\t0\t144\n",
      "19\t144\t3526.5\t0\t144\n",
      "20\t105\t3526.5\t0\t105\n",
      "21\t464\t3526.5\t0\t464\n",
      "22\t82\t3526.5\t0\t82\n",
      "23\t91\t3526.5\t0\t91\n",
      "24\t153\t3526.5\t0\t153\n",
      "25\t113\t3526.5\t0\t113\n",
      "26\t75\t3526.5\t0\t75\n",
      "27\t124\t3526.5\t0\t124\n",
      "28\t80\t3526.5\t0\t80\n",
      "29\t85\t3526.5\t0\t85\n",
      "30\t179\t3526.5\t0\t179\n",
      "31\t72\t3526.5\t0\t72\n",
      "32\t161\t3526.5\t0\t161\n",
      "33\t191\t3526.5\t0\t191\n",
      "34\t132\t3526.5\t0\t132\n",
      "35\t105\t3526.5\t0\t105\n",
      "36\t58\t3526.5\t0\t58\n",
      "37\t25\t3526.5\t0\t25\n",
      "38\t30\t3526.5\t0\t30\n",
      "39\t29\t3526.5\t0\t29\n",
      "40\t22\t3526.5\t0\t22\n",
      "41\t90\t3526.5\t0\t90\n",
      "42\t32\t3526.5\t0\t32\n",
      "43\t34\t3526.5\t0\t34\n",
      "44\t46\t3526.5\t0\t46\n",
      "45\t35\t3526.5\t0\t35\n",
      "46\t40\t3526.5\t0\t40\n",
      "47\t8\t3526.5\t0\t8\n",
      "48\t22\t3526.5\t0\t22\n",
      "49\t2\t3526.5\t0\t2\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "input_list_file=\"sample_list.txt\"\n",
    "while IFS='' read -r line || [[ -n \"$line\" ]]; do\n",
    "  sample_list=\"$sample_list $line\"\n",
    "done <\"$input_list_file\"\n",
    "echo $sample_list\n",
    "\n",
    "for read in $sample_list; do\n",
    "    cat 'merged_'$read'.fastq' \"unmerged_\"$read\"_R1.fastq\" > $read'.fastq'\n",
    "\n",
    "\n",
    "    cutadapt -j 12 \\\n",
    "        --discard-untrimmed \\\n",
    "        --minimum-length 20 \\\n",
    "        -a CTGTAGGCACCATCAATGAC \\\n",
    "        -e 0.1 \\\n",
    "        -o 'dt_'$read'.fastq' $read'.fastq'\n",
    "\n",
    "\n",
    "    move_umi.py -r 'dt_'$read'.fastq' \\\n",
    "                -l 4 \\\n",
    "                -o 'umiRemoved_'\"$read\" \\\n",
    "                -i ATCACG \\\n",
    "                -a ATCACGCTGTAGGCACCATCAATGAC \\\n",
    "                -m 'pipeline'\n",
    "\n",
    "    cutadapt -j 12 \\\n",
    "        --discard-untrimmed \\\n",
    "        --minimum-length 20 \\\n",
    "        -a ATCACG \\\n",
    "        -e 0.01 \\\n",
    "        -o 'trimmed_'$read'.fastq' 'umiRemoved_'$read'_R1.fastq'\n",
    "\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Process with just Read1 only"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Alignment\n",
    "\n",
    "`STAR` was used to perform sequence alignment against curated *Leishmania* rRNA sequences.   "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "May 11 23:04:07 ..... started STAR run\n",
      "May 11 23:04:07 ..... loading genome\n",
      "May 11 23:04:07 ..... started mapping\n",
      "May 11 23:04:34 ..... finished mapping\n",
      "May 11 23:04:35 ..... started sorting BAM\n",
      "May 11 23:04:52 ..... finished successfully\n",
      "May 11 23:04:53 ..... started STAR run\n",
      "May 11 23:04:53 ..... loading genome\n",
      "May 11 23:04:53 ..... started mapping\n",
      "May 11 23:05:16 ..... finished mapping\n",
      "May 11 23:05:16 ..... started sorting BAM\n",
      "May 11 23:05:33 ..... finished successfully\n",
      "May 11 23:05:34 ..... started STAR run\n",
      "May 11 23:05:34 ..... loading genome\n",
      "May 11 23:05:34 ..... started mapping\n",
      "May 11 23:05:51 ..... finished mapping\n",
      "May 11 23:05:52 ..... started sorting BAM\n",
      "May 11 23:06:02 ..... finished successfully\n",
      "May 11 23:06:06 ..... started STAR run\n",
      "May 11 23:06:06 ..... loading genome\n",
      "May 11 23:06:06 ..... started mapping\n",
      "May 11 23:06:06 ..... finished mapping\n",
      "May 11 23:06:06 ..... finished successfully\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "input_list_file=\"sample_list.txt\"\n",
    "while IFS='' read -r line || [[ -n \"$line\" ]]; do\n",
    "  sample_list=\"$sample_list $line\"\n",
    "done <\"$input_list_file\"\n",
    "for sample in $sample_list; do\n",
    "    mkdir -p $sample\n",
    "    STAR --runThreadN 10 \\\n",
    "      --outMultimapperOrder Random \\\n",
    "      --outFilterMismatchNoverLmax 0.1 \\\n",
    "      --alignEndsType EndToEnd \\\n",
    "      --limitOutSJcollapsed 20000000 \\\n",
    "      --limitIObufferSize 550000000 \\\n",
    "      --limitBAMsortRAM 2960722171 \\\n",
    "      --genomeDir $GENOMES\"/LM/\" \\\n",
    "      --readFilesIn 'trimmed_'\"$sample\"'.fastq' \\\n",
    "      --outFileNamePrefix ./\"$sample\"/ \\\n",
    "      --genomeLoad LoadAndKeep \\\n",
    "      --outSAMtype BAM SortedByCoordinate\n",
    "#       --outFilterScoreMinOverLread 0.8 \\\n",
    "#       --outFilterMatchNminOverLread 0.8 \\\n",
    "done\n",
    "\n",
    "STAR --genomeDir $GENOMES\"/LM/\" --genomeLoad Remove"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Post alignment processing\n",
    "\n",
    "`samtools` was used to index the alignment output.  \n",
    "`umi_tools` was used to remove PCR duplicates.  \n",
    "`bedtools` was used to generate human readable bed files for downstream analysis.  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# UMI-tools version: 1.0.1\n",
      "# output generated by dedup -I Aligned.sortedByCoord.out.bam -S LM1_dedup.bam\n",
      "# job started at Tue May 11 23:07:43 2021 on yzylhome -- 8135461f-fd54-494c-8f89-31ea84f213bb\n",
      "# pid: 8184, system: Linux 5.4.0-73-generic #82-Ubuntu SMP Wed Apr 14 17:39:42 UTC 2021 x86_64\n",
      "# assigned_tag                            : None\n",
      "# cell_tag                                : None\n",
      "# cell_tag_delim                          : None\n",
      "# cell_tag_split                          : -\n",
      "# chimeric_pairs                          : use\n",
      "# chrom                                   : None\n",
      "# compresslevel                           : 6\n",
      "# detection_method                        : None\n",
      "# gene_tag                                : None\n",
      "# gene_transcript_map                     : None\n",
      "# get_umi_method                          : read_id\n",
      "# ignore_umi                              : False\n",
      "# in_sam                                  : False\n",
      "# log2stderr                              : False\n",
      "# loglevel                                : 1\n",
      "# mapping_quality                         : 0\n",
      "# method                                  : directional\n",
      "# no_sort_output                          : False\n",
      "# out_sam                                 : False\n",
      "# output_unmapped                         : False\n",
      "# paired                                  : False\n",
      "# per_cell                                : False\n",
      "# per_contig                              : False\n",
      "# per_gene                                : False\n",
      "# random_seed                             : None\n",
      "# read_length                             : False\n",
      "# short_help                              : None\n",
      "# skip_regex                              : ^(__|Unassigned)\n",
      "# soft_clip_threshold                     : 4\n",
      "# spliced                                 : False\n",
      "# stats                                   : False\n",
      "# stderr                                  : <_io.TextIOWrapper name='<stderr>' mode='w' encoding='utf-8'>\n",
      "# stdin                                   : <_io.TextIOWrapper name='Aligned.sortedByCoord.out.bam' mode='r' encoding='UTF-8'>\n",
      "# stdlog                                  : <_io.TextIOWrapper name='<stdout>' mode='w' encoding='utf-8'>\n",
      "# stdout                                  : <_io.TextIOWrapper name='LM1_dedup.bam' mode='w' encoding='UTF-8'>\n",
      "# subset                                  : None\n",
      "# threshold                               : 1\n",
      "# timeit_file                             : None\n",
      "# timeit_header                           : None\n",
      "# timeit_name                             : all\n",
      "# tmpdir                                  : None\n",
      "# umi_sep                                 : _\n",
      "# umi_tag                                 : RX\n",
      "# umi_tag_delim                           : None\n",
      "# umi_tag_split                           : None\n",
      "# unmapped_reads                          : discard\n",
      "# unpaired_reads                          : use\n",
      "# whole_contig                            : False\n",
      "2021-05-11 23:07:43,288 INFO command: dedup -I Aligned.sortedByCoord.out.bam -S LM1_dedup.bam\n",
      "2021-05-11 23:08:11,953 INFO Parsed 1000000 input reads\n",
      "2021-05-11 23:08:13,399 INFO Parsed 2000000 input reads\n",
      "2021-05-11 23:08:15,000 INFO Parsed 3000000 input reads\n",
      "2021-05-11 23:08:20,689 INFO Parsed 4000000 input reads\n",
      "2021-05-11 23:08:47,170 INFO Written out 100000 reads\n",
      "2021-05-11 23:08:48,065 INFO Parsed 5000000 input reads\n",
      "2021-05-11 23:08:49,046 INFO Parsed 6000000 input reads\n",
      "2021-05-11 23:08:50,335 INFO Parsed 7000000 input reads\n",
      "2021-05-11 23:09:27,382 INFO Parsed 8000000 input reads\n",
      "2021-05-11 23:09:28,617 INFO Parsed 9000000 input reads\n",
      "2021-05-11 23:09:30,078 INFO Parsed 10000000 input reads\n",
      "2021-05-11 23:09:30,098 INFO Written out 200000 reads\n",
      "2021-05-11 23:09:30,452 INFO Parsed 11000000 input reads\n",
      "2021-05-11 23:09:32,636 INFO Parsed 12000000 input reads\n",
      "2021-05-11 23:09:37,014 INFO Reads: Input Reads: 12428626\n",
      "2021-05-11 23:09:37,014 INFO Number of reads out: 222357\n",
      "2021-05-11 23:09:37,014 INFO Total number of positions deduplicated: 6108\n",
      "2021-05-11 23:09:37,014 INFO Mean number of unique UMIs per position: 161.00\n",
      "2021-05-11 23:09:37,014 INFO Max. number of unique UMIs per position: 280\n",
      "# job finished in 113 seconds at Tue May 11 23:09:37 2021 -- 113.84  1.47  0.00  0.00 -- 8135461f-fd54-494c-8f89-31ea84f213bb\n",
      "# UMI-tools version: 1.0.1\n",
      "# output generated by dedup -I Aligned.sortedByCoord.out.bam -S LM4_dedup.bam\n",
      "# job started at Tue May 11 23:09:40 2021 on yzylhome -- f0863c19-f1f1-4b5f-8f84-3fd546f8af6d\n",
      "# pid: 8214, system: Linux 5.4.0-73-generic #82-Ubuntu SMP Wed Apr 14 17:39:42 UTC 2021 x86_64\n",
      "# assigned_tag                            : None\n",
      "# cell_tag                                : None\n",
      "# cell_tag_delim                          : None\n",
      "# cell_tag_split                          : -\n",
      "# chimeric_pairs                          : use\n",
      "# chrom                                   : None\n",
      "# compresslevel                           : 6\n",
      "# detection_method                        : None\n",
      "# gene_tag                                : None\n",
      "# gene_transcript_map                     : None\n",
      "# get_umi_method                          : read_id\n",
      "# ignore_umi                              : False\n",
      "# in_sam                                  : False\n",
      "# log2stderr                              : False\n",
      "# loglevel                                : 1\n",
      "# mapping_quality                         : 0\n",
      "# method                                  : directional\n",
      "# no_sort_output                          : False\n",
      "# out_sam                                 : False\n",
      "# output_unmapped                         : False\n",
      "# paired                                  : False\n",
      "# per_cell                                : False\n",
      "# per_contig                              : False\n",
      "# per_gene                                : False\n",
      "# random_seed                             : None\n",
      "# read_length                             : False\n",
      "# short_help                              : None\n",
      "# skip_regex                              : ^(__|Unassigned)\n",
      "# soft_clip_threshold                     : 4\n",
      "# spliced                                 : False\n",
      "# stats                                   : False\n",
      "# stderr                                  : <_io.TextIOWrapper name='<stderr>' mode='w' encoding='utf-8'>\n",
      "# stdin                                   : <_io.TextIOWrapper name='Aligned.sortedByCoord.out.bam' mode='r' encoding='UTF-8'>\n",
      "# stdlog                                  : <_io.TextIOWrapper name='<stdout>' mode='w' encoding='utf-8'>\n",
      "# stdout                                  : <_io.TextIOWrapper name='LM4_dedup.bam' mode='w' encoding='UTF-8'>\n",
      "# subset                                  : None\n",
      "# threshold                               : 1\n",
      "# timeit_file                             : None\n",
      "# timeit_header                           : None\n",
      "# timeit_name                             : all\n",
      "# tmpdir                                  : None\n",
      "# umi_sep                                 : _\n",
      "# umi_tag                                 : RX\n",
      "# umi_tag_delim                           : None\n",
      "# umi_tag_split                           : None\n",
      "# unmapped_reads                          : discard\n",
      "# unpaired_reads                          : use\n",
      "# whole_contig                            : False\n",
      "2021-05-11 23:09:40,655 INFO command: dedup -I Aligned.sortedByCoord.out.bam -S LM4_dedup.bam\n",
      "2021-05-11 23:10:11,418 INFO Parsed 1000000 input reads\n",
      "2021-05-11 23:10:12,859 INFO Parsed 2000000 input reads\n",
      "2021-05-11 23:10:14,409 INFO Parsed 3000000 input reads\n",
      "2021-05-11 23:10:20,772 INFO Parsed 4000000 input reads\n",
      "2021-05-11 23:10:47,843 INFO Written out 100000 reads\n",
      "2021-05-11 23:10:48,370 INFO Parsed 5000000 input reads\n",
      "2021-05-11 23:10:49,331 INFO Parsed 6000000 input reads\n",
      "2021-05-11 23:10:50,563 INFO Parsed 7000000 input reads\n",
      "2021-05-11 23:11:37,729 INFO Parsed 8000000 input reads\n",
      "2021-05-11 23:11:38,950 INFO Parsed 9000000 input reads\n",
      "2021-05-11 23:11:40,156 INFO Parsed 10000000 input reads\n",
      "2021-05-11 23:11:40,943 INFO Parsed 11000000 input reads\n",
      "2021-05-11 23:11:41,021 INFO Parsed 12000000 input reads\n",
      "2021-05-11 23:11:41,244 INFO Written out 200000 reads\n",
      "2021-05-11 23:11:43,338 INFO Parsed 13000000 input reads\n",
      "2021-05-11 23:11:47,785 INFO Reads: Input Reads: 13394914\n",
      "2021-05-11 23:11:47,785 INFO Number of reads out: 213916\n",
      "2021-05-11 23:11:47,785 INFO Total number of positions deduplicated: 6108\n",
      "2021-05-11 23:11:47,785 INFO Mean number of unique UMIs per position: 160.05\n",
      "2021-05-11 23:11:47,785 INFO Max. number of unique UMIs per position: 297\n",
      "# job finished in 127 seconds at Tue May 11 23:11:47 2021 -- 127.03  1.45  0.00  0.00 -- f0863c19-f1f1-4b5f-8f84-3fd546f8af6d\n",
      "# UMI-tools version: 1.0.1\n",
      "# output generated by dedup -I Aligned.sortedByCoord.out.bam -S LM5_dedup.bam\n",
      "# job started at Tue May 11 23:11:50 2021 on yzylhome -- cd870821-951e-46a5-b2f0-46ca3742923c\n",
      "# pid: 8245, system: Linux 5.4.0-73-generic #82-Ubuntu SMP Wed Apr 14 17:39:42 UTC 2021 x86_64\n",
      "# assigned_tag                            : None\n",
      "# cell_tag                                : None\n",
      "# cell_tag_delim                          : None\n",
      "# cell_tag_split                          : -\n",
      "# chimeric_pairs                          : use\n",
      "# chrom                                   : None\n",
      "# compresslevel                           : 6\n",
      "# detection_method                        : None\n",
      "# gene_tag                                : None\n",
      "# gene_transcript_map                     : None\n",
      "# get_umi_method                          : read_id\n",
      "# ignore_umi                              : False\n",
      "# in_sam                                  : False\n",
      "# log2stderr                              : False\n",
      "# loglevel                                : 1\n",
      "# mapping_quality                         : 0\n",
      "# method                                  : directional\n",
      "# no_sort_output                          : False\n",
      "# out_sam                                 : False\n",
      "# output_unmapped                         : False\n",
      "# paired                                  : False\n",
      "# per_cell                                : False\n",
      "# per_contig                              : False\n",
      "# per_gene                                : False\n",
      "# random_seed                             : None\n",
      "# read_length                             : False\n",
      "# short_help                              : None\n",
      "# skip_regex                              : ^(__|Unassigned)\n",
      "# soft_clip_threshold                     : 4\n",
      "# spliced                                 : False\n",
      "# stats                                   : False\n",
      "# stderr                                  : <_io.TextIOWrapper name='<stderr>' mode='w' encoding='utf-8'>\n",
      "# stdin                                   : <_io.TextIOWrapper name='Aligned.sortedByCoord.out.bam' mode='r' encoding='UTF-8'>\n",
      "# stdlog                                  : <_io.TextIOWrapper name='<stdout>' mode='w' encoding='utf-8'>\n",
      "# stdout                                  : <_io.TextIOWrapper name='LM5_dedup.bam' mode='w' encoding='UTF-8'>\n",
      "# subset                                  : None\n",
      "# threshold                               : 1\n",
      "# timeit_file                             : None\n",
      "# timeit_header                           : None\n",
      "# timeit_name                             : all\n",
      "# tmpdir                                  : None\n",
      "# umi_sep                                 : _\n",
      "# umi_tag                                 : RX\n",
      "# umi_tag_delim                           : None\n",
      "# umi_tag_split                           : None\n",
      "# unmapped_reads                          : discard\n",
      "# unpaired_reads                          : use\n",
      "# whole_contig                            : False\n",
      "2021-05-11 23:11:50,898 INFO command: dedup -I Aligned.sortedByCoord.out.bam -S LM5_dedup.bam\n",
      "2021-05-11 23:12:13,473 INFO Parsed 1000000 input reads\n",
      "2021-05-11 23:12:15,736 INFO Parsed 2000000 input reads\n",
      "2021-05-11 23:12:22,161 INFO Parsed 3000000 input reads\n",
      "2021-05-11 23:12:37,732 INFO Written out 100000 reads\n",
      "2021-05-11 23:12:38,084 INFO Parsed 4000000 input reads\n",
      "2021-05-11 23:12:43,151 INFO Parsed 5000000 input reads\n",
      "2021-05-11 23:13:25,279 INFO Parsed 6000000 input reads\n",
      "2021-05-11 23:13:26,166 INFO Parsed 7000000 input reads\n",
      "2021-05-11 23:13:27,422 INFO Parsed 8000000 input reads\n",
      "2021-05-11 23:13:27,745 INFO Parsed 9000000 input reads\n",
      "2021-05-11 23:13:28,023 INFO Parsed 10000000 input reads\n",
      "2021-05-11 23:13:35,260 INFO Written out 200000 reads\n",
      "2021-05-11 23:13:36,443 INFO Reads: Input Reads: 10821953\n",
      "2021-05-11 23:13:36,443 INFO Number of reads out: 201168\n",
      "2021-05-11 23:13:36,443 INFO Total number of positions deduplicated: 6097\n",
      "2021-05-11 23:13:36,443 INFO Mean number of unique UMIs per position: 151.75\n",
      "2021-05-11 23:13:36,443 INFO Max. number of unique UMIs per position: 288\n",
      "# job finished in 105 seconds at Tue May 11 23:13:36 2021 -- 105.49  1.38  0.00  0.00 -- cd870821-951e-46a5-b2f0-46ca3742923c\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# Read in sample from out list and store as a variable:\n",
    "input_list_file=\"sample_list.txt\"\n",
    "while IFS='' read -r line || [[ -n \"$line\" ]]; do\n",
    "  sample_list=\"$sample_list $line\"\n",
    "done <\"$input_list_file\"\n",
    "for sample in $sample_list; do\n",
    "  cd \"$sample\"\n",
    "  samtools index \"Aligned.sortedByCoord.out.bam\"\n",
    "  umi_tools dedup -I \"Aligned.sortedByCoord.out.bam\" -S $sample'_dedup.bam'\n",
    "  bedtools bamtobed -i $sample'_dedup.bam' >'../bed_files/SE_aligned/'$sample'.bed'\n",
    "  cd ../\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
